backend/app/api/routes/gemini_generate.py

```
@@ -0,0 +1,39 @@
+    }
+    headers = {
+        "Content-Type": "application/json",
+    }
+    params = {"key": GEMINI_API_KEY}
+    try:
+        async with httpx.AsyncClient(timeout=30.0) as client:
+            response = await client.post(GEMINI_API_URL, json=payload, headers=headers, params=params)
+        response.raise_for_status()
+        return response.json()
+    except httpx.RequestError as e:
+        raise HTTPException(status_code=502, detail=f"Gemini API error: {str(e)}")
```

Ruff (0.14.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)